flytekit/annotated/base_task.py

```diff
@@ -33,6 +35,57 @@ def kwtypes(**kwargs) -> Dict[str, Type]:
     return d
 
 
+@dataclass
+class TaskMetadata(object):
+    """
+    Create Metadata to be associated with this Task
+
+    Args:
+      cache: Boolean that indicates if caching should be enabled
+      cache_version: Version string to be used for the cached value
+      interruptible: Boolean that indicates that this task is of for interruptions and can be scheduled on nodes
+                           with lower QoS guarantees. This will directly reduce the `$`/`execution cost` associated,
+                            at the cost of performance penalties due to potential interruptions
+      deprecated: This should be used to indicate that the task is deprecated
```

Comment:
nit: can you provide suggestions for what the value should be? at first glance i would wonder why it's not a bool

Reply:
its the acutal deprecated message, i kept it as is, but will add a comment